syntax = "proto3";
package rendezqueue;

message TrySwapRequest {
  bytes key = 1;
  uint32 id = 2;
  uint32 ttl = 3;
  uint32 offset = 4;
  repeated bytes values = 5;
}

message TrySwapResponse {
  // Your key. Just to confirm. Can be ignored.
  bytes key = 1;
  // Your id. Just to confirm. Can be ignored.
  uint32 id = 2;
  // Remaining seconds.
  uint32 ttl = 3;
  // Holds number of values (when values is not returned).
  uint32 offset = 4;
  // Other party's values.
  repeated bytes values = 5;
}

service RendezqueueService {
  rpc TrySwap (TrySwapRequest) returns (TrySwapResponse) {}
}
